src/Sentry.AzureFunctions.Worker/SentryFunctionsWorkerMiddleware.cs

```diff
@@ -0,0 +1,82 @@
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Middleware;
+
+namespace Sentry.AzureFunctions.Worker;
+
+internal class SentryFunctionsWorkerMiddleware : IFunctionsWorkerMiddleware
+{
+    private readonly IHub _hub;
+
+    public SentryFunctionsWorkerMiddleware(IHub hub)
+    {
+        _hub = hub;
+    }
+
+    public async Task Invoke(FunctionContext context, FunctionExecutionDelegate next)
+    {
+        try
+        {
+            var transactionName = await GetTransactionNameAsync(context) ?? context.FunctionDefinition.Name;
+
+            _hub.ConfigureScope(scope =>
+            {
+                var transaction = _hub.StartTransaction(transactionName, "function");
+                scope.Transaction = transaction;
+
+                scope.UnsetTag("AzureFunctions_FunctionName");
+                scope.UnsetTag("AzureFunctions_InvocationId");
+
+                scope.Contexts["function"] = new Dictionary<string, string>
+                {
+                    { "name", context.FunctionDefinition.Name },
+                    { "entryPoint", context.FunctionDefinition.EntryPoint },
+                    { "invocationId", context.InvocationId }
+                };
+            });
+
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            await next(context).ConfigureAwait(false);
+        }
+        catch (Exception exception)
+        {
+            exception.SetSentryMechanism(nameof(SentryFunctionsWorkerMiddleware),
+                "This exception was caught by the Sentry Functions middleware. " +
+                "The Function has thrown an exception that was not handled by the user code.",
+                handled: false);
+
+            throw;
+        }
+    }
+
+    private static async Task<string?> GetTransactionNameAsync(FunctionContext context)
+    {
+        // Get the HTTP request data
+        var requestData = await context.GetHttpRequestDataAsync().ConfigureAwait(false);
+        if (requestData is null)
+        {
+            // not an HTTP trigger
+            return null;
+        }
+
+        // Find the HTTP Trigger attribute via reflection
+        var assembly = Assembly.LoadFrom(context.FunctionDefinition.PathToAssembly);
+        var entryPointName = context.FunctionDefinition.EntryPoint;
+
+        var typeName = entryPointName[..entryPointName.LastIndexOf('.')];
+        var methodName = entryPointName[(typeName.Length + 1)..];
+        var attribute = assembly.GetType(typeName)?.GetMethod(methodName)?.GetParameters()
+            .Select(p => p.GetCustomAttribute<HttpTriggerAttribute>())
+            .FirstOrDefault(a => a is not null);
```

Comment:
Since we have no choice but to use reflection - we should probably keep a cache so we don't have to do it repeatedly.

Reply:
I didn't bother with that as I wanted to see if anything was coming soon. Since it's not, I'll add something, 👍

Reply:
Regarding the cached transaction name and the lookup key. Suppose I go with a `context.FunctionDefinition.PathToAssembly` + `context.FunctionDefinition.EntryPoint` + HTTP method combo; the result will look something like this:

```
<path>\sentry-dotnet\samples\Sentry.Samples.AzureFunctions.Worker\bin\Debug\net6.0\Sentry.Samples
.AzureFunctions.Worker.dll-Sentry.Samples.AzureFunctions.Worker.SampleHttpTrigger.RunAsync-GET
```

This feels somewhat redundant, @mattjohnsonpint. I think `context.FunctionDefinition.EntryPoint` + HTTP method combo should suffice, which will generate 

```
Sentry.Samples.AzureFunctions.Worker.SampleHttpTrigger.RunAsync-GET
```

`Sentry.Samples.AzureFunctions.Worker` - the assembly name containing the trigger
`SampleHttpTrigger.RunAsync` - the class name + the entry method
`GET` - the HTTP method.

What do you think, @mattjohnsonpint?

Reply:
That's fine. Just go with the entrypoint. We can always adjust it later if we get any complaints.